macros/schema_tests/aggregate_functions/expect_column_most_common_value_to_be_in_set.sql

```diff
@@ -22,6 +22,8 @@
                                                                       row_condition
                                                                       ) %}
 
+{% set data_type = dbt.type_numeric() if not data_type else data_type %}
```

Comment:
fyi, Snowflake won't correctly compare a float in the db to a value cast to decimal (default for the test) so makes more sense to use the built-in types across the test to make sure comparisons actually work.

Reply:
From [the official documentation](https://docs.snowflake.com/en/sql-reference/data-types-numeric) and my testing,  inserting floating point data will create a `number` column type with the smallest precision and scale to fit the value. e.g.:  when creating a table with initial data, if a column contains [0, 1, 0.5], the assigned type will be `number(2,1)`. So no information is lost at that point. However casting that to "numeric", "decimal", "number" type without explicit precision and scale will result in rounding and casting it to any "float" or "double" type [has the usual accuracy problems](https://en.wikipedia.org/wiki/Floating-point_arithmetic#Accuracy_problems).

When comparing, Snowflake will happily attempt to compare any numeric type with the usual issues from floats
 <table>
  <tr>
    <th>comparison</th>
    <th>result</th>
  </tr>
  <tr>
   <td>cast(0.5 as number(2,1)) = cast(0.5 as numeric(28,2))</td>
   <td>TRUE</td>
     </tr>
  <tr>
   <td>cast(0.5 as float) = cast(0.5 as numeric(28,2))</td>
   <td>TRUE</td>
     </tr>
  <tr>
    <td>cast(0.5 as float) = cast(0.5 as numeric)</td>
    <td>FALSE</td>
     </tr>
<tr>
    <td>0.1 + 0.2 = 0.3</td>
    <td>TRUE</td>
  </tr>
<tr>
    <td>(cast(0.1 as float) + cast(0.2 as float)) = cast(0.3 as float)</td>
    <td>FALSE</td>
  </tr>
  <tr>
    <td>cast(3.1415926535897932384626433832795028841 as number(38,37)) = cast(3.141592653589793 as float)</td>
    <td>TRUE</td>
  </tr>
</table> 

*TL;DR*: It makes sense to use built-in types everywhere to not get db-engine specific issues.